wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/spi/RuntimeInfrastructure.java

```diff
@@ -162,7 +94,6 @@ public RuntimeContext prepare(RuntimeIdentity id, InternalEnvironment environmen
     for (InternalEnvironmentProvisioner provisioner : internalEnvironmentProvisioners) {
```

Comment:
IMO it is too abstract to understand what's going on
Why not to add environment variables needed for all the machines just on creation InternalEnvironment with factory?

Reply:
It also adds environment variables from installers to the machine where they are. And moves servers from installers to the machine. This is why it is a bit abstract. 
+1 to move into Factory for now. In case we decide to add estimation into workspace API later we would be willing to move this back since it is not a part of environment recipe processing.

Reply:
I agree to move it into InternalEnvironmentFactory for now

Reply:
@gazarenkov @garagatyi I have a problem with such refactoring. WorkpaceIdEnvVar provisioner and MachineTokenEnvVarProvisioner need workspace id to provide the corresponding environment variable. It requires a refactoring and this PR already has a ton of changes. Are you OK to fix it separately in another issue?
P.S. I think It's OK since this PR doesn't introduce this functionality.

Reply:
+1 to avoid adding new changes not related to this PR

Reply:
OK, let's do it in next PR.
Just to make sure - I am talking about removing "provisioner" abstraction on SPI level at all (before GA of course).
Choose for it whatever tactic you want.